internal/provider/custom_domain_verification_resource.go

```diff
@@ -209,3 +211,32 @@ func (r *customDomainVerificationResource) Update(ctx context.Context, req resou
 func (r *customDomainVerificationResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
 	tflog.Trace(ctx, "deleted a custom_domain resource")
 }
+
+func (r *customDomainVerificationResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	var state customDomainVerificationResourceModel
+
+	environment := req.ID
+	if environment != "Staging" && environment != "Prod" {
+		resp.Diagnostics.AddError("Invalid import ID", "The import ID must be either `Staging` or `Prod`.")
+		return
+	}
+
+	customDomainInfo, err := r.client.GetCustomDomainInfo(environment, false)
+	if err != nil {
+		resp.Diagnostics.AddError("Error fetching custom domain info", "Could not fetch custom domain info for the environment.")
+		return
+	}
+
+	state.Environment = types.StringValue(environment)
+	state.Domain = types.StringValue(customDomainInfo.Domain)
+	// need to manually pull timeouts from hcl to pass validation
+	var timeouts timeouts.Value
```

Comment:
Pulled the workaround for importing timeouts manually from the tf team's suggestion [here](https://github.com/hashicorp/terraform-plugin-framework-timeouts/issues/46).